packages/camera/camera_android_camerax/lib/src/camera_control.dart

```diff
@@ -158,11 +171,20 @@ class _CameraControlHostApiImpl extends CameraControlHostApi {
       CameraControl instance, int index) async {
     final int identifier = instanceManager.getIdentifier(instance)!;
     try {
-      return setExposureCompensationIndex(identifier, index);
+      final int? exposureCompensationIndex =
+          await setExposureCompensationIndex(identifier, index);
+      if (exposureCompensationIndex == null) {
+        SystemServices.cameraErrorStreamController.add(
+            'Setting exposure compensation index was canceled due to the camera being closed or a new request being submitted.');
+        return Future<int?>.value();
+      }
+      return exposureCompensationIndex;
     } on PlatformException catch (e) {
       SystemServices.cameraErrorStreamController.add(e.message ??
           'Setting the camera exposure compensation index failed.');
-      return Future<int?>.value();
+      // Surfacing error to plugin layer to maintain consistency of
+      // setExposureOffset implementation across platform implementations.
+      rethrow;
```

Comment:
Looking for feedback on this as per https://github.com/flutter/packages/pull/6059/files#r1481841622.